types/leaflet/index.d.ts

```diff
@@ -1821,17 +1821,17 @@ export interface GeoJSONOptions<P = any> extends InteractiveLayerOptions {
  * Represents a GeoJSON object or an array of GeoJSON objects.
  * Allows you to parse GeoJSON data and display it on the map. Extends FeatureGroup.
  */
-export class GeoJSON<P = any> extends FeatureGroup<P> {
+export class GeoJSON<P = any, G extends geojson.GeometryObject = geojson.GeometryObject> extends FeatureGroup<P> {
     /**
      * Convert layer into GeoJSON feature
      */
-    static getFeature<P = any>(layer: Layer, newGeometry: geojson.Feature<geojson.GeometryObject, P> | geojson.GeometryObject): geojson.Feature<geojson.GeometryObject, P>;
+    static getFeature<P = any, G extends geojson.GeometryObject = geojson.GeometryObject>(layer: Layer, newGeometry: geojson.Feature<G, P> | G): geojson.Feature<G, P>;
 
     /**
      * Creates a Layer from a given GeoJSON feature. Can use a custom pointToLayer
      * and/or coordsToLatLng functions if provided as options.
      */
-    static geometryToLayer<P = any>(featureData: geojson.Feature<geojson.GeometryObject, P>, options?: GeoJSONOptions<P>): Layer;
+    static geometryToLayer(featureData: geojson.Feature, options?: GeoJSONOptions): Layer;
```

Comment:
Why does this drop the type parameters to Feature?

Reply:
I think this is a mistake. I originally added geometry as type parameter, but then removed both because they don't appear in the return type. I see now that they matter because it ties the function parameters together.

These changes are in a separate commit, so I'll rebase and remove it to keep the commit history clean.